packages/test/src/test-integration-workflows.ts

```diff
@@ -216,6 +216,49 @@ test('Condition 0 patch sets a timer', async (t) => {
   t.false(await worker.runUntil(executeWorkflow(conditionTimeout0)));
 });
 
+export async function historySizeBytesGrows(): Promise<[number, number]> {
+  const before = workflow.workflowInfo().historySizeBytes;
+  await workflow.sleep(1);
+  const after = workflow.workflowInfo().historySizeBytes;
+  return [before, after];
+}
+
+test('HistorySizeBytes grows with new WFT', async (t) => {
+  const { createWorker, executeWorkflow } = helpers(t);
+  const worker = await createWorker();
+  const [before, after] = await worker.runUntil(executeWorkflow(historySizeBytesGrows));
+  t.assert(after > before && before > 100);
+});
+
+test('HistorySizeBytes is visible in WorkflowExecutionInfo', async (t) => {
+  const { createWorker, startWorkflow } = helpers(t);
+  const worker = await createWorker();
+  const handle = await startWorkflow(historySizeBytesGrows);
+
+  await worker.runUntil(handle.result());
+  const historySizeBytes = (await handle.describe()).historySizeBytes;
+  t.assert(historySizeBytes > 100);
+});
+
+export async function suggestedCAN(): Promise<boolean> {
+  const MAX_EVENTS = 40_000;
+  const BATCH_SIZE = 100;
+  while (workflow.workflowInfo().historyLength < MAX_EVENTS) {
+    await Promise.all(new Array(BATCH_SIZE).fill(undefined).map((_) => workflow.sleep(1)));
```

Comment:
nit: I'd use an upsert search attributes instead, it's cheaper to run.

Reply:
It is actually pretty fast, and even against the cloud, it will succeed in just 20 round trips (the actual server settings  are 4K events, not 40K)